packages/client/src/open-feature.ts

```diff
@@ -190,15 +191,25 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI<Provider, Hook> impleme
   }
 
   private async runProviderContextChangeHandler(
+    clientName: string | undefined,
     provider: Provider,
     oldContext: EvaluationContext,
     newContext: EvaluationContext,
   ): Promise<void> {
-    try {
-      return await provider.onContextChange?.(oldContext, newContext);
-    } catch (err) {
-      this._logger?.error(`Error running ${provider.metadata.name}'s context change handler:`, err);
-    }
+      const providerName = provider.metadata.name;
+      return await provider.onContextChange?.(oldContext, newContext).then(() => {
```

Comment:
You shouldn't need to await this. Is there any reason you're using a Promise chain instead of async/await?

Reply:
Removed the await.

I just prefer this to try/catch in situations like this. I like seeing returns close to the top of methods/functions.